### PR TITLE
feat: Add health check endpoint

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/CAT-345

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,10 @@ export N8N_RUNNERS_AUTH_TOKEN=...
 make run
 ```
 
+## Health check
+
+The launcher exposes a health check endpoint at `/healthz` on port `5681` for liveness checks.
```
Comment (ivov) [inline README.md:115]: Do we want this configurable? Or too early?

File: internal/http/healthcheck_server.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	res := struct {
+		Status string `json:"status"`
+	}{Status: "ok"}
+
+	if err := json.NewEncoder(w).Encode(res); err != nil {
+		logs.Logger.Printf("Failed to encode health check response: %v", err)
```
Comment (tomi) [inline internal/http/healthcheck_server.go:46]: Shouldn't we log with error level?

File: README.md [hunk 3]
```diff
@@ -110,6 +110,12 @@ export N8N_RUNNERS_AUTH_TOKEN=...
 make run
 ```
 
+## Health check
+
+The launcher exposes a health check endpoint at `/healthz` on port `5681` for liveness checks. To override the launcher health check port, set the `N8N_LAUNCHER_HEALTCHECK_PORT` env var. 
+
+If overriding the default health check port, be mindful of port conflicts - the n8n main instance uses `5678` by default for its HTTP server and `5679` for its WS server, and the runner uses `5680` by default for its WS server.
```
Comment (tomi) [inline README.md:117]:
```suggestion
If overriding the default health check port, be mindful of port conflicts - the n8n main instance uses `5678` by default for its HTTP server and `5679` for its task runner server, and the runner uses `5680` by default for its healthcheck server. 
```

**Review by tomi — COMMENTED** [review]
Nice! 🙌 Only couple tiny things

File: internal/http/healthcheck_server.go (hunk 2 above)
Comment (ivov) [reply internal/http/healthcheck_server.go]: Yeah back then this was waiting for the logging PR to be merged, same below in `GetPort` 👍🏻

**Review by tomi — APPROVED** [review]
🚀

**Pull request merged**